SIGSEV when re-creating mittwald_virtualhost resource

```
╷
│ Error: Request cancelled
│
│ The plugin6.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-mittwald plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x102b39ad8]

goroutine 8 [running]:
github.com/mittwald/terraform-provider-mittwald/internal/provider/resource/virtualhostresource.(*Resource).Create(0x140002c5d00, {0x102f97928, 0x140003bcb70}, {{{{0x102f9d990, 0x140003dd1d0}, {0x102e26100, 0x140003bde60}}, {0x102fa0458, 0x140002aa000}}, {{{0x102f9d990, ...}, ...}, ...}, ...}, ...)
	/Users/mhelmich/Git/Github/terraform-provider-mittwald/internal/provider/resource/virtualhostresource/resource.go:129 +0x288
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0x140002506c8, {0x102f97928, 0x140003bcb70}, 0x140002bd508, 0x140002bd4e0)
	/Users/mhelmich/Git/Go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_createresource.go:101 +0x3fc
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0x140002506c8, {0x102f97928, 0x140003bcb70}, 0x140003b4960, 0x140002bd600)
	/Users/mhelmich/Git/Go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_applyresourcechange.go:57 +0x380
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0x140002506c8, {0x102f97928?, 0x140003bca80?}, 0x140003b48c0)
	/Users/mhelmich/Git/Go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/proto6server/server_applyresourcechange.go:55 +0x2dc
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x14000259040, {0x102f97928?, 0x140003bc2d0?}, 0x140003ba300)
	/Users/mhelmich/Git/Go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.27.0/tfprotov6/tf6server/server.go:989 +0x294
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x102f6aec0, 0x14000259040}, {0x102f97928, 0x140003bc2d0}, 0x140003ba280, 0x0)
	/Users/mhelmich/Git/Go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.27.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:687 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400024c200, {0x102f97928, 0x140003bc240}, 0x140002812c0, 0x140003b3500, 0x10368c888, 0x0)
	/Users/mhelmich/Git/Go/pkg/mod/google.golang.org/grpc@v1.72.1/server.go:1405 +0xc9c
google.golang.org/grpc.(*Server).handleStream(0x1400024c200, {0x102f98310, 0x14000400000}, 0x140002812c0)
	/Users/mhelmich/Git/Go/pkg/mod/google.golang.org/grpc@v1.72.1/server.go:1815 +0x900
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	/Users/mhelmich/Git/Go/pkg/mod/google.golang.org/grpc@v1.72.1/server.go:1035 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 36
	/Users/mhelmich/Git/Go/pkg/mod/google.golang.org/grpc@v1.72.1/server.go:1046 +0x138

Error: The terraform-provider-mittwald plugin crashed!
```

--- a/internal/provider/resource/virtualhostresource/resource.go
+++ b/internal/provider/resource/virtualhostresource/resource.go
@@ -125,6 +125,9 @@ func (r *Resource) Create(ctx context.Context, req resource.CreateRequest, resp 
 		ingress := providerutil.
 			Try[*domainclientv2.CreateIngressResponse](&resp.Diagnostics, "API error while creating virtual host").
 			DoValResp(client.CreateIngress(ctx, data.ToCreateRequest(ctx, &resp.Diagnostics)))
+		if resp.Diagnostics.HasError() {
+			return
+		}
 
 		data.ID = types.StringValue(ingress.Id)
 	}

